src/AlgoliaSearchCore.js

```diff
@@ -360,19 +360,20 @@ AlgoliaSearchCore.prototype._jsonRequest = function(initialOpts) {
         return client._promise.reject(err);
       }
 
-      if (err instanceof errors.RequestTimeout) {
+      if (err instanceof errors.RequestTimeout || err instanceof errors.NodeNetwork) {
         return retryRequest();
       } else if (!usingFallback) {
-        client.hostIndex[initialOpts.hostType] = ++client.hostIndex[initialOpts.hostType] % client.hosts[initialOpts.hostType].length;
         // next request loop, force using fallback for this request
         tries = Infinity;
       }
 
+      client.hostIndex[initialOpts.hostType] = (client.hostIndex[initialOpts.hostType] + 1) % client.hosts[initialOpts.hostType].length;
+
       return doRequest(requester, reqOpts);
     }
 
     function retryRequest() {
-      client.hostIndex[initialOpts.hostType] = ++client.hostIndex[initialOpts.hostType] % client.hosts[initialOpts.hostType].length;
+      client.hostIndex[initialOpts.hostType] = (client.hostIndex[initialOpts.hostType] + 1) % client.hosts[initialOpts.hostType].length;
```

Comment:
From what I understand, not incrementing that `hostIndex` here will result in a loop always using the same host in case of `RequestTimeout` or `NodeNetwork`; is that correct?